service/src/main/kotlin/app/cash/backfila/ui/pages/ServiceIndexAction.kt

```diff
@@ -0,0 +1,208 @@
+package app.cash.backfila.ui.pages
+
+import app.cash.backfila.ui.PathBuilder
+import app.cash.backfila.ui.components.DashboardLayout
+import app.cash.backfila.ui.components.PageTitle
+import app.cash.backfila.ui.components.ServiceAutocompleteWrapper
+import app.cash.backfila.ui.components.ToggleContainer
+import javax.inject.Inject
+import kotlinx.html.a
+import kotlinx.html.div
+import kotlinx.html.form
+import kotlinx.html.id
+import kotlinx.html.input
+import kotlinx.html.legend
+import misk.hotwire.buildHtml
+import misk.security.authz.Unauthenticated
+import misk.turbo.turbo_frame
+import misk.web.Get
+import misk.web.QueryParam
+import misk.web.ResponseContentType
+import misk.web.actions.WebAction
+import misk.web.mediatype.MediaTypes
+
+class ServiceIndexAction @Inject constructor() : WebAction {
+  @Get(PATH)
+  @ResponseContentType(MediaTypes.TEXT_HTML)
+  @Unauthenticated
+  fun get(
+    @QueryParam sc: String?,
+  ): String {
+    return buildHtml {
+      DashboardLayout(
+        title = "Backfila",
+        path = PATH,
+      ) {
+        PageTitle("Services")
+        val pathBuilder = PathBuilder(path = PATH)
+
+        ServiceAutocompleteWrapper(redirectPath = ServiceShowAction.PATH)
+
+        ToggleContainer(
+          buttonText = "${pathBuilder.countFilters()} Filters",
+          labelBlock = {
+            a(classes = "text-gray-500") {
+              href = PathBuilder(
+                path = pathBuilder.path,
+                service = pathBuilder.service,
+                // sortKey = pathBuilder.sortKey,
```

Comment:
Do we want these?

Reply:
I might need them later, will cleanup in follow up.